### PR TITLE
Fix windows build

### LINKED ISSUE #54: Tests fail on wheels for windows
I discovered this bug while adjusting the wheel tests performed by cibuiltwheel (previously weren't run on built wheels)

**Describe the bug**
CIBuildWheel wheels are failing unit tests on Windows. For now, I've disabled wheel tests on windows.

**To Reproduce**
Re-enable Windows wheel tests by deleting this line.
https://github.com/emilydolson/phylotrackpy/blob/2d4015d3fc26e2f320cb84f505849bbc261ff706/.github/workflows/ci.yml#L65

**Expected behavior**
Wheel tests should pass on Windows.

**Computational environment (please complete the following information):**
 - OS: windows
 - Python version: Cpython 3.6
 
**Additional context**
Here is the full error from log https://github.com/emilydolson/phylotrackpy/actions/runs/6986563896/job/19012150190

```
  ================================== FAILURES ===================================
  ___________________________ test_systematics[taxa0] ___________________________
  
  taxa = ['hello', 'hello 2']
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_systematics(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
          org_tax = systematics.Taxon(0, tax1)
  >       org2_tax = sys.add_org(org2, org_tax)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:65: RuntimeError
  ___________________________ test_systematics[taxa1] ___________________________
  
  taxa = [1, 2]
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_systematics(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
          org_tax = systematics.Taxon(0, tax1)
  >       org2_tax = sys.add_org(org2, org_tax)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:65: RuntimeError
  ___________________________ test_systematics[taxa2] ___________________________
  
  taxa = [1.0, 2.0]
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_systematics(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
          org_tax = systematics.Taxon(0, tax1)
  >       org2_tax = sys.add_org(org2, org_tax)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:65: RuntimeError
  ___________________________ test_systematics[taxa3] ___________________________
  
  taxa = [[1], [1, 2]]
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_systematics(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
          org_tax = systematics.Taxon(0, tax1)
  >       org2_tax = sys.add_org(org2, org_tax)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:65: RuntimeError
  ___________________________ test_systematics[taxa4] ___________________________
  
  taxa = [array([1]), array([1, 2])]
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_systematics(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
          org_tax = systematics.Taxon(0, tax1)
  >       org2_tax = sys.add_org(org2, org_tax)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:65: RuntimeError
  _______________________ test_taxa_serialization[taxa0] ________________________
  
  taxa = ['hello', 'hello 2']
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_taxa_serialization(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
  >       org_tax = sys.add_org(org)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:99: RuntimeError
  _______________________ test_taxa_serialization[taxa1] ________________________
  
  taxa = [1, 2]
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_taxa_serialization(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
  >       org_tax = sys.add_org(org)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:99: RuntimeError
  _______________________ test_taxa_serialization[taxa2] ________________________
  
  taxa = [1.0, 2.0]
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_taxa_serialization(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
  >       org_tax = sys.add_org(org)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:99: RuntimeError
  _______________________ test_taxa_serialization[taxa3] ________________________
  
  taxa = [[1], [1, 2]]
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_taxa_serialization(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
  >       org_tax = sys.add_org(org)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:99: RuntimeError
  _______________________ test_taxa_serialization[taxa4] ________________________
  
  taxa = [array([1]), array([1, 2])]
  
      @mark.parametrize(
          "taxa",
          (
              ["hello", "hello 2"],
              [1, 2],
              [1.0, 2.0],
              [[1], [1, 2]],
              [np.array([1]), np.array([1, 2])],
          ),
      )
      def test_taxa_serialization(taxa):
          tax1, tax2 = taxa
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org = ExampleOrg(tax1)
          org2 = ExampleOrg(tax2)
  >       org_tax = sys.add_org(org)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:99: RuntimeError
  ____________________________ test_shared_ancestor _____________________________
  
      def test_shared_ancestor():
          sys = systematics.Systematics(taxon_info_fun, True, True, False, False)
          org1 = ExampleOrg("hello")
          org2 = ExampleOrg("hello 2")
          org3 = ExampleOrg("hello 3")
          org4 = ExampleOrg("hello 4")
      
  >       org1_tax = sys.add_org(org1)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:123: RuntimeError
  ____________________________ test_phylostatistics _____________________________
  
      def test_phylostatistics():
          sys = systematics.Systematics(str, True, True, False, False)
      
          sys.set_update(0)
  >       id1 = sys.add_org(25)
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:157: RuntimeError
  ________________________________ test_deepcopy ________________________________
  
      def test_deepcopy():
          sys = systematics.Systematics(lambda x: x, True, True, False, False)
  >       tax = sys.add_org("hello")
  E       RuntimeError: Internal Error (in D:\a\phylotrackpy\phylotrackpy\Empirical/include/emp/Evolve/Systematics.hpp line 1458): !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.",
  E       !store_position && "Trying to add org to position-tracking systematics manager without position. Either specify a valid position or turn of position tracking for systematic manager.": [1]
  
  D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py:209: RuntimeError
  =========================== short test summary info ===========================
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_systematics[taxa0]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_systematics[taxa1]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_systematics[taxa2]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_systematics[taxa3]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_systematics[taxa4]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_taxa_serialization[taxa0]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_taxa_serialization[taxa1]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_taxa_serialization[taxa2]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_taxa_serialization[taxa3]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_taxa_serialization[taxa4]
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_shared_ancestor
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_phylostatistics
  FAILED D:\a\phylotrackpy\phylotrackpy\test\test_systematics.py::test_deepcopy
  ======================== 13 failed, 5 passed in 6.47s =========================
Error: Command python -m pytest --import-mode=importlib D:\a\phylotrackpy\phylotrackpy/test failed with code 1. None
```

### PR DESCRIPTION
**Pull request by emilydolson**
Attempting to fix #54

### CODE REVIEW

**Comment by emilydolson** [comment]
It works! Had to gut optional _throw a bit (we should eventually try to add back the functionality I just removed), but I'd say it's functional.

If you're good with removing NumPy as a dependency, feel free to merge this. Also happy to be convinced to keep numpy

**Comment by mmore500** [comment]
I suspect that numpy-data-as-taxon-info is a potential use case. I'm thinking of how DEAP has a whole section on Numpy genetic representations <https://deap.readthedocs.io/en/master/tutorials/advanced/numpy.html>

If reverting back NumPy breaks tests, though, let's axe it.

**Comment by emilydolson** [comment]
How does this look to you? It tests the numpy functionality without actually adding it as a dependency or testing it in the wheels (where build systems get messier)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
+        pip install numpy
```
Comment (mmore500) [inline .github/workflows/ci.yml]:
```suggestion
        pip install oldest-supported-numpy==2023.10.25
```

**Review by mmore500 — CHANGES_REQUESTED** [review]
👍 this setup seems like a good idea

If I’m reading the diffs right, we still need to uncomment numpy test cases and add the `nowheel` marker you created?

Here’s the syntax for that btw <https://docs.pytest.org/en/7.1.x/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test>

Added a suggestion to pin numpy install on the test job. I think there’s an argument to be made either way (pin or track latest). The argument to pin being that (potentially very confusing) breakage would occur when you’re working on something unrelated and potentially cause a wild goose chase or annoyingly couple a task to unrelated maintenance issues.

(Pinning numpy directly doesn’t work well b/c wheels aren’t available across all python versions, hence oldest-supported-numpy).

**Comment by emilydolson** [comment]
I believe the numpy test case is already be marked with the nowheel and running. Pinning numpy is good with me (although I think it should be less fragile in the test environment than in the wheel environment)

**Comment by mmore500** [comment]
Ah, I see that one of the tests is set up, put the other one together in f55acab. Merging!

**Pull request merged**